src/server.h

```diff
@@ -141,6 +143,21 @@ struct ClientInfo {
 	std::string vers_string, lang_code;
 };
 
+struct ModIPCStore {
+	/// RW lock for this entire structure
+	std::shared_mutex mutex;
+	/// Signalled on any changes to the map contents
+	std::condition_variable_any condvar;
+	/**
+	 * Map storing the data
+	 *
+	 * Note: Do not store `nil` data in this map, instead remove the whole key.
+	 */
+	std::unordered_map<std::string, std::unique_ptr<PackedValue>> map;
+
+	inline void signal() { condvar.notify_all(); }
```

Comment:
why add a wrapper around this one specific function? `wait_until` is invoked directly anyways and it doesn't save on LOC.

Reply:
I think my idea was that it's not obvious which of `notify_all` or `notify` should be called.